src/fe/fe_hermite.C

```diff
@@ -272,41 +273,9 @@ unsigned int hermite_n_dofs_per_elem(const ElemType t,
 } // anonymous namespace
 
 
-
-
-  // Do full-specialization of nodal_soln() function for every
-  // dimension, instead of explicit instantiation at the end of this
-  // file.
-  // This could be macro-ified so that it fits on one line...
-template <>
-void FE<0,HERMITE>::nodal_soln(const Elem * elem,
-                               const Order order,
-                               const std::vector<Number> & elem_soln,
-                               std::vector<Number> & nodal_soln)
-{ hermite_nodal_soln(elem, order, elem_soln, nodal_soln); }
-
-template <>
-void FE<1,HERMITE>::nodal_soln(const Elem * elem,
-                               const Order order,
-                               const std::vector<Number> & elem_soln,
-                               std::vector<Number> & nodal_soln)
-{ hermite_nodal_soln(elem, order, elem_soln, nodal_soln); }
-
-template <>
-void FE<2,HERMITE>::nodal_soln(const Elem * elem,
-                               const Order order,
-                               const std::vector<Number> & elem_soln,
-                               std::vector<Number> & nodal_soln)
-{ hermite_nodal_soln(elem, order, elem_soln, nodal_soln); }
-
-template <>
-void FE<3,HERMITE>::nodal_soln(const Elem * elem,
-                               const Order order,
-                               const std::vector<Number> & elem_soln,
-                               std::vector<Number> & nodal_soln)
-{ hermite_nodal_soln(elem, order, elem_soln, nodal_soln); }
-
-
+// Instantiate (side_) nodal_soln() function for every dimension
+LIBMESH_FE_NODAL_SOLN(HERMITE, hermite_nodal_soln)
```

Comment:
This PR seems to have some less related FE refactoring in it, e.g. macro-izing `FE::nodal_soln` instantiations, that might have worked well as a separate PR, but probably too painful to split up at this point.